project-management-openproject/project-management-openproject-api/src/main/java/com/xwiki/projectmanagement/openproject/internal/rest/connection/CreateConnection.java

```diff
@@ -42,27 +43,35 @@
  */
 @Component
 @Named("com.xwiki.projectmanagement.openproject.internal.rest.connection.CreateConnection")
-@Path("openproject/connections")
+@Path("wikis/{wikiName}/spaces/{spaceName:.+}/pages/{pageName}/openproject/configurations")
 public class CreateConnection extends XWikiResource
 {
     @Inject
     private CreateConnectionService createConnectionService;
+
     /**
-     * @param data dasdsads
-     * @return things.
+     * Creates a new OpenProject connection configuration document in XWiki using the provided path parameters and
+     * configuration data.
+     *
+     * @param wikiName the name of the wiki where the connection page will be created
+     * @param spaceName the full space path where the page resides
+     * @param pageName the name of the page where the connection configuration will be stored
+     * @param openProjectConnection the {@link OpenProjectConnection} to store
+     * @return a response indicating the result of the operation
      */
     @POST
-    @Path("/create")
     @Consumes({ MediaType.APPLICATION_JSON })
     @Produces({ MediaType.APPLICATION_JSON })
-    public Response createConnection(Map<String, Object> data)
+    public Response createConnection(
+        @PathParam("wikiName") String wikiName,
+        @PathParam("spaceName") String spaceName,
+        @PathParam("pageName") String pageName,
+        OpenProjectConnection openProjectConnection)
     {
-        String connectionName = (String) data.get("connectionName");
-        String serverURL = (String) data.get("serverURL");
-        String clientId = (String) data.get("clientId");
-        String clientSecret = (String) data.get("clientSecret");
         try {
-            createConnectionService.createConnection(connectionName, serverURL, clientId, clientSecret);
+            DocumentReference documentReference =
```

Comment:
I just realized that you should check that the current user has Admin rights before executing the connection service.